macro/src/lib.rs

```diff
@@ -33,25 +33,53 @@ struct AttrArgs {
     key_expr: Expr,
     store_type: Option<Type>,
     store_init: Option<Expr>,
+    result: Option<bool>,
+    option: Option<bool>,
 }
 fn obtain_return_type(return_type: ReturnType) -> Type {
     match return_type {
         syn::ReturnType::Type(_, return_type) => *return_type,
         syn::ReturnType::Default => unimplemented!("default return types are not supported"),
     }
 }
+fn extract_inner_type(typ: Type) -> Type {
```

Comment:
We suppose this function will be changed due to removal of `Option` support?

Reply:
Does it need to? It functions correctly for `Result` types already and I'm not aware of a better way to extract the inner type of a `Result`.